provider/bidengine/order.go

```diff
@@ -360,6 +360,13 @@ loop:
 
 	// cancel reservation
 	if !won {
+		if clusterch != nil {
+			result := <-clusterch
```

Comment:
This makes sense but we also need to avoid blocking forever here. The simplest solution may just be to use `time.After` with a long timeout. @boz thoughts?

Reply:
Can there be a case where this channel will block infinitely? I looked into `func (is *inventoryService) reserve` code and this always seems to return data to the channel.

Reply:
We try to be really careful about this stuff.  Since we're not cancelling the context I worry a bit, but if it's a clear case then okay, forget it.

Alternatively we can send in context that has a bound duration to begin with.

Reply:
@boz if the inventory service is shutting down it won't answer any more requests right? So wouldn't this goroutine get stuck "forever" when stopping the provider?

Reply:
It *should* respond with an error when it's shutting down.  Let me check.

Reply:
Does this cover the case you're talking about @hydrogen18? https://github.com/ovrclk/akash/blob/5df45f7b009e04b95dbdf430bf5f149b9cab14c7/provider/cluster/inventory.go#L155-L157